### PR TITLE
treewide: add version tests

### PR DESCRIPTION
**Pull request by khaneliman**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/darwin/sketchybar/default.nix [hunk 1]
```diff
@@ -53,6 +55,11 @@ stdenv.mkDerivation (finalAttrs: {
     runHook postInstall
   '';
 
+  passthru.tests.version = testers.testVersion {
+    package = sketchybar;
```
Comment (Artturin) [inline pkgs/os-specific/darwin/sketchybar/default.nix:59]: When finalAttrs is available then `package = finalAttrs.finalPackage;`

File: pkgs/os-specific/darwin/skhd/default.nix [hunk 2]
```diff
@@ -31,6 +33,11 @@ stdenv.mkDerivation (finalAttrs: {
     substituteInPlace $out/Library/LaunchDaemons/org.nixos.skhd.plist --subst-var out
   '';
 
+  passthru.tests.version = testers.testVersion {
+    package = skhd;
```
Comment (Artturin) [inline pkgs/os-specific/darwin/skhd/default.nix:37]: finalAttrs available

File: pkgs/tools/misc/ncdu/default.nix [hunk 3]
```diff
@@ -31,6 +33,11 @@ stdenv.mkDerivation (finalAttrs: {
     installManPage ncdu.1
   '';
 
+  passthru.tests.version = testers.testVersion {
+    package = ncdu;
```
Comment (Artturin) [inline pkgs/tools/misc/ncdu/default.nix:37]: finalAttrs available

File: pkgs/applications/misc/1password/default.nix [hunk 4]
```diff
@@ -63,6 +63,11 @@ stdenv.mkDerivation {
     $out/bin/${mainProgram} --version
   '';
 
+  passthru.tests.version = testers.testVersion {
+    package = _1password;
+    version = "${version}";
```
Comment (figsoda) [inline pkgs/applications/misc/1password/default.nix:68]: this is the default
```suggestion
```

File: pkgs/tools/misc/lazydocker/default.nix [hunk 5]
```diff
@@ -21,6 +21,11 @@ buildGoModule rec {
 
   ldflags = [ "-s" "-w" "-X main.version=${version}" ];
 
+  passthru.tests.version = testers.testVersion {
+    package = lazydocker;
+    version = "Version: ${version}";
```
Comment (figsoda) [inline pkgs/tools/misc/lazydocker/default.nix:26]: I think we can just use the default here
```suggestion
```

File: pkgs/tools/misc/ncdu/default.nix [hunk 6]
```diff
@@ -31,6 +32,11 @@ stdenv.mkDerivation (finalAttrs: {
     installManPage ncdu.1
   '';
 
+  passthru.tests.version = testers.testVersion {
+    package = finalAttrs.finalPackage;
+    version = "ncdu ${finalAttrs.version}";
```
Comment (figsoda) [inline pkgs/tools/misc/ncdu/default.nix:37]:
```suggestion
```

File: pkgs/tools/misc/toilet/default.nix [hunk 7]
```diff
@@ -12,6 +12,11 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [ pkg-config ];
   buildInputs = [ libcaca ];
 
+  passthru.tests.version = testers.testVersion {
+    package = toilet;
+    version = "Version: ${version}";
```
Comment (figsoda) [inline pkgs/tools/misc/toilet/default.nix:17]:
```suggestion
```

File: pkgs/tools/system/bottom/default.nix [hunk 8]
```diff
@@ -36,6 +38,11 @@ rustPlatform.buildRustPackage rec {
 
   BTM_GENERATE = true;
 
+  passthru.tests.version = testers.testVersion {
+    package = bottom;
+    version = "bottom ${version}";
```
Comment (figsoda) [inline pkgs/tools/system/bottom/default.nix:43]:
```suggestion
```

File: pkgs/tools/system/btop/default.nix [hunk 9]
```diff
@@ -30,6 +32,11 @@ stdenv.mkDerivation rec {
     ${removeReferencesTo}/bin/remove-references-to -t ${stdenv.cc.cc} $(readlink -f $out/bin/btop)
   '';
 
+  passthru.tests.version = testers.testVersion {
+    package = btop;
+    version = "btop version: ${version}";
```
Comment (figsoda) [inline pkgs/tools/system/btop/default.nix:37]:
```suggestion
```

Comment (figsoda) [inline pkgs/tools/system/btop/default.nix:37]:
```suggestion
```

File: pkgs/tools/misc/lazydocker/default.nix (hunk 5 above)
Comment (khaneliman) [reply pkgs/tools/misc/lazydocker/default.nix:26]: oh so this doesn't have to be super strict string matching? it just searches for the version inside the output?

Comment (figsoda) [reply pkgs/tools/misc/lazydocker/default.nix:26]: Yes, it searches full words that matches `version`

**Comment by khaneliman** [comment]
@figsoda @Artturin looks like checks just finished after latest changes.

**Pull request merged**